plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java

```diff
@@ -46,9 +94,86 @@ public String getDescription() {
   }
 
   @Override
-  public IFile createProjectFiles(IProject newProject, AppEngineProjectConfig config, IProgressMonitor monitor)
+  public IFile createAndConfigureProjectContent(IProject newProject, AppEngineProjectConfig config,
+      IProgressMonitor monitor) throws CoreException {
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+    IFile mostImportantFile =  CodeTemplates.materializeAppEngineFlexFiles(newProject, config,
+        subMonitor.newChild(30));
+    configureFacets(newProject, subMonitor.newChild(20));
+    addDependenciesToProject(newProject, subMonitor.newChild(50));
+    return mostImportantFile;
+  }
+
+  /**
+   * Add Java 8 and Dynamic Web Module facet
+   */
+  private void configureFacets(IProject project, IProgressMonitor monitor) throws CoreException {
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+    IFacetedProject facetedProject = ProjectFacetsManager.create(
+        project, true /* convertIfNecessary */, subMonitor.newChild(50));
+    Set<IFacetedProject.Action> facetInstallSet = new HashSet<>();
+    FacetUtil.addJavaFacetToBatch(JavaFacet.VERSION_1_8, facetedProject, facetInstallSet);
+    FacetUtil.addWebFacetToBatch(WebFacetUtils.WEB_30, facetedProject, facetInstallSet);
+    FacetUtil.addFacetSetToProject(facetedProject, facetInstallSet, subMonitor.newChild(50));
+  }
+
+  private void addDependenciesToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    return CodeTemplates.materializeAppEngineFlexFiles(newProject, config, monitor);
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+
+    // Create a lib folder
+    IFolder libFolder = project.getFolder("lib");
+    if (!libFolder.exists()) {
+      libFolder.create(true, true, subMonitor.newChild(10));
+    }
+
+    // Download the dependencies from maven
+    int ticks = 50 / PROJECT_DEPENDENCIES.size();
+    for (Map.Entry<String, String> dependency : PROJECT_DEPENDENCIES.entrySet()) {
+      LibraryFile libraryFile = new LibraryFile(new MavenCoordinates(dependency.getKey(),
+          dependency.getValue()));
+      Artifact artifact = null;
+      try {
+        artifact = repositoryService.resolveArtifact(libraryFile, subMonitor.newChild(ticks));
+      } catch (CoreException ex) {
```

Comment:
Looking at this logic, I believe it's better to merge this `try` block and the other block below IMO: it's either that you succeed to copy the artifact or not. (Basically, I am talking about Advantage 1 of the exception mechanism in [here](http://docs.oracle.com/javase/tutorial/essential/exceptions/advantages.html).) What you are doing now seems just like the traditional error code checking (i.e., using `if` to check a return code of a function).